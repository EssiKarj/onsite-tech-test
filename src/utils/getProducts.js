export default function getProducts() {
  return new Promise(resolve => {
    const products = [
      {
        id: 1,
        title: 'Personalised cufflinks',
        image: 'cufflinks.jpg',
        altText: 'cufflinks',
        price: '45.00',
      },
      {
        id: 2,
        title: 'Kids T-shirt',
        image: 'tshirt.jpg',
        altText: 'kids t-shirt',
        price: '19.95',
      },
    ];

    setTimeout(() => resolve(products), 1000);
  });
}
