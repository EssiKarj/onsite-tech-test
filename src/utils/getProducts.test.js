import getProducts from './getProducts';

test('returns a list of products', async () => {
  const productList = await getProducts();

  expect(productList.length).toBe(2);
  expect(productList).toMatchSnapshot();
});
