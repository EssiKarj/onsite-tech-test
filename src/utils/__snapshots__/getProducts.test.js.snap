// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns a list of products 1`] = `
Array [
  Object {
    "altText": "cufflinks",
    "id": 1,
    "image": "cufflinks.jpg",
    "price": "45.00",
    "title": "Personalised cufflinks",
  },
  Object {
    "altText": "kids t-shirt",
    "id": 2,
    "image": "tshirt.jpg",
    "price": "19.95",
    "title": "Kids T-shirt",
  },
]
`;
